subscription/src/main/java/org/killbill/billing/subscription/engine/dao/DefaultSubscriptionDao.java

```diff
@@ -513,34 +513,57 @@ public Void inTransaction(final EntitySqlDaoWrapperFactory entitySqlDaoWrapperFa
     }
 
     @Override
-    public void createSubscriptionWithAddOns(final List<DefaultSubscriptionBase> subscriptions, final Map<UUID, List<SubscriptionBaseEvent>> initialEventsMap, final InternalCallContext context) {
+    public void createSubscriptionsWithAddOns(final List<List<DefaultSubscriptionBase>> subscriptionsWithAddOns,
```

Comment:
Do we really need to group these as a `List<List<>>`? Or could we simplify it through the entire call chain and flatten it as a simple list?